resteasy-reactor/src/main/java/org/jboss/resteasy/reactor/MonoRxInvokerImpl.java

```diff
@@ -3,146 +3,158 @@
 import jakarta.ws.rs.client.Entity;
 import jakarta.ws.rs.core.GenericType;
 import jakarta.ws.rs.core.Response;
-
 import org.jboss.resteasy.client.jaxrs.internal.ClientInvocationBuilder;
 import org.jboss.resteasy.client.jaxrs.internal.PublisherRxInvokerImpl;
+import org.reactivestreams.Publisher;
 import reactor.core.publisher.Mono;
 
+import java.util.Objects;
 import java.util.concurrent.CompletionStage;
 
-public class MonoRxInvokerImpl extends PublisherRxInvokerImpl implements MonoRxInvoker {
+public class MonoRxInvokerImpl implements MonoRxInvoker {
 
-    public MonoRxInvokerImpl(final ClientInvocationBuilder builder) {
-        super(builder);
+
+    private final MonoPublisherInvoker monoInvoker;
+
+
+    static class MonoPublisherInvoker extends PublisherRxInvokerImpl {
+        MonoPublisherInvoker(final ClientInvocationBuilder builder) {
+            super(builder);
+        }
+
+        @Override
+        protected <T> Publisher<T> toPublisher(CompletionStage<T> completable) {
+            return Mono.fromFuture(completable.toCompletableFuture());
```

Comment:
Probably not critical, but shouldn't `Mono.fromCompletionStage(completable)` work here?

Reply:
fixed it in next commit